plugins/gatsby-remark-ample/README.md

```diff
@@ -347,3 +348,36 @@ The intention here was to not remove the original field reference to the SEO dat
   }
 }
 ```
+
+## Plugins
+
+This plugin accepts an array of plugin dependencies. Each dependency is expected to have a `remark-plugin.js` file in its root directory. That file is to contain a series of exported APIs that enable the plugin to hook into the lifecycle of the node creation process provided by this plugin.
+
+For example, say you want to add a `url` field to objects with the `Post` object before it is created. If that logic were to be included in a `gatsby-plugin-posts` plugin, you'd first specify the dependency in `gatsby-config.js`:
+
+```js
+{
+  resolve: `gatsby-remark-ample`,
+  options: {
+    plugins: [`gatsby-plugin-posts`]
```

Comment:
This may be best as a future iteration but how would you envision allowing for configuration of one of the plugins.

1. Would we just edit the plugin? (I don't really foresee a lot of going back and updating plugins so this may be ok)
1. Allow passing in an object or a string? (I think the Gatsby plugin API handles it this way.)
1. Other?

Reply:
This is a Gatsby pattern — using `plugins` as an option to other plugins. It comes as a rich object by the time the plugin has access to the array.

Should we want to support options to this dependencies, we would pass objects here, like we do with [`gatsby-transformer-remark`](https://github.com/ample/gatsby-starter-ample/blob/main/gatsby-config.js#L90-L110). I believe this _just works_ with Gatsby.